docs/ARCHITECTURE.md

```diff
@@ -351,14 +351,14 @@ direction LR
         }
 
         class MessageBuilder {
-            +usingRole(MessageRole $role) self
+            +usingRole(MessageRoleEnum $role) self
+            +usingUserRole() self
+            +usingModelRole() self
             +withText(string $text) self
-            +withImageFile(File $file) self
-            +withAudioFile(File $file) self
-            +withVideoFile(File $file) self
+            +withFile($file, ?string $mimeType) self
```

Comment:
Great catch updating these! Since you're already on it, could you revise the `PromptBuilder` definition here as well, at least replacing similar outdated instances?